Handle Slackware OS version strings containing a plus (“+”)

edrozenberg opened the pull request:
A couple of years ago Slackware -current began using a plus (“+”) at the end of the distribution version string to indicate the os distribution being used is an in-progress/future version. No-plus: stable. Plus: in-progress

##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Slackware distribution version strings can contain a plus ("+") when the Slackware OS being used is the in-progress/future release version.

For example:
Slackware stable OS distribution string: `Slackware 14.2`
Slackware -current (in-progress) OS distribution string: `Slackware 14.2+`

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
Fixes https://github.com/ansible/ansible/issues/38760

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
ansible/module_utils/facts/system/distribution.py

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
`ansible -m setup HOST_RUNNING_SLACKWARE_CURRENT | grep ansible_distribution_version`

<!--- Paste verbatim command output below, e.g. before and after your change -->
Before change:
```
"ansible_distribution_version": "14.2",
```

After change:
```
"ansible_distribution_version": "14.2+",
```

edrozenberg committed:
Handle Slackware OS version strings containing a plus (“+”)

A couple of years ago Slackware -current began using a plus (“+”) at the end of the distribution version string to indicate a future version work-in-progress.

bcoca commented:
Could you add this to the test cases for distribution fact gathering?

samdoran commented:
This change seems ok. My only concern was how having a `+` character in the version would mess with our `version` test but it seems to mostly do the correct thing. The only test that fails is `'14.2+' is version('14.2', '==')`, which I would expect.

Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

Also, it looks like we don't have any fixture data for Slackware in our tests. Could you run [gen_distribution_version.py](hacking/tests/gen_distribution_version_testcase.py) on some Slackware systems and add the test fixture files [here](test/units/module_utils/facts/system/distribution/fixtures)?

samdoran reviewed (changes requested):
The change looks good, we just need a changelog and test data.

edrozenberg commented:
> The change looks good, we just need a changelog and test data.

Is that a request for me to provide this info? I don't work for Ansible BTW, just a guy who's trying to get Ansible to report the correct OS version :).  Let me know and I'll look into what's being required.

samdoran committed:
Rearrange distribution_files unit tests to easily support more tests

- add conftest with common fixtures
- use parametrize for testing multiple scenarios

samdoran committed:
Add changelog

samdoran committed:
Add unit tests for Slackware distribution parsing

samdoran committed:
Use correct fixtures for Slackware

Data comes from /etc/slackware-version
